### DISCUSSION TITLE
Multiple outputs (SOLVED)

### DISCUSSION THREAD

**Discussion by HunLucas** [rern/rAudio#529] [General]
Hello everyone!

I want to use a sound sensitive led driver with my raspberry, and this need a JACK connection. I need help to setup the outputs.
I tried some methods but not work.

Here is my aplay -l output
`aplay -l
**** List of PLAYBACK Hardware Devices ****
card 0: sndrpihifiberry [snd_rpi_hifiberry_dacplus], device 0: HiFiBerry DAC+ Pro HiFi pcm512x-hifi-0 [HiFiBerry DAC+ Pro HiFi pcm512x-hifi-0]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: Device [USB PnP Sound Device], device 0: USB Audio [USB Audio]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
`

1.
Manually add a sound card to mpd.conf, the OS always writes over, and clean up my config.
audio_output {
	name           "HiFiBerry DAC+ Pro"
	device         "hw:0,0"
	type           "alsa"
	auto_resample  "no"
	auto_format    "no"
	mixer_type     "hardware"
	mixer_control  "Digital"
	mixer_device   "hw:0"
}

audio_output {
	name           "USB PnP Sound Device"
	device         "hw:1,0"
	type           "alsa"
	auto_resample  "no"
	auto_format    "no"
	mixer_type     "hardware"
	mixer_control  "PCM"
	mixer_device   "hw:1"
}

2.  .asoundrc config.
pcm.multi {
    type multi;
    slaves.a.pcm "hw:0,0";
    slaves.a.channels 2;
    slaves.b.pcm "hw:1,0";
    slaves.b.channels 2;
    bindings.0.slave a;
    bindings.0.channel 0;
    bindings.1.slave a;
    bindings.1.channel 1;
    bindings.2.slave b;
    bindings.2.channel 0;
    bindings.3.slave b;
    bindings.3.channel 1;
}

ctl.multi {
    type hw;
    card 0;
}

pcm.ttable {
    type route;
    slave.pcm "multi";
    ttable.0.0 1;
    ttable.1.1 1;
    ttable.2.2 1;
    ttable.3.3 1;
}
ctl.ttable {
    type hw;
    card 0;
}
After this I have Json error on runeaudio player setting page
`Errors: Unexpected token d in JSON at position 566
{ "page" : "player" , "devices" : [{ "aplayname" : "hifiberry-dacplus" , "card" : 0 , "device" : 0 , "dop" : 0 , "hw" : "hw:0,0" , "hwmixer" : "Digital" , "mixers" : 1 , "mixerdevices" : ["Digital"] , "mixermanual" : false , "mixertype" : "software" , "name" : "HiFiBerry DAC+ Pro" },{ "aplayname" : "USB PnP Sound Device" , "card" : 1 , "device" : 0 , "dop" : 0 , "hw" : "hw:1,0" , "hwmixer" : "PCM" , "mixers" : 1 , "mixerdevices" : ["PCM"] , "mixermanual" : false , "mixertype" : "hardware" , "name" : "USB PnP Sound Device" }] , "active" : true , "asoundcard" : ▮defaults.pcm.card1 , "audioaplayname" : "USB PnP Sound Device" , "audiooutput" : "" , "autoupdate" : false , "buffer" : false , "bufferval" : false , "bufferoutput" : false , "bufferoutputval" : false , "counts" : { "album": 2630, "albumartist": 1685, "artist": 1156, "composer": 117, "conductor": 0, "date": 158, "genre": 146, "nas": 0, "sd": 0, "usb": 1, "song": 12455, "webradio": 4 } , "crossfade" : false , "crossfadeval" : false , "custom" : false , "ffmpeg" : false , "normalization" : false , "reboot" : "" , "replaygain" : false , "replaygainval" : "" , "soxr" : false , "soxrval" : "" , "version" : "0.22.11-1"}`

If someone have a solution, please tell me :)

Thank You!

**Comment by rern** [comment]
`mpd.conf`
Insert in global section, after `user   "mpd"` line.

Or

User's Configuration:
- Lines in the upper box (global section) will be maintained across outputs and restarts.
- Lines in the lower box (within output part) will be appended to current output. Each output can have it's own custom lines.

The error:
- **rAudio** creates and reads current audio card number from the first line of `/etc/asound.conf`
- `/etc/asound.conf` will be overwritten on restart.

**Comment by HunLucas** [comment]
I made a clean install, and now its works. Next time I ask before I change the config files... Thank you very much!
